main/views.py

```diff
@@ -114,4 +121,5 @@ def user_forgot_password(request):
 
 @login_required
 def game(request):
+    create_UserProfile(request, request.user)
```

Comment:
Can we do this some other way, avoid calling the function that creates the User Profile everytime the Game page accessed? Increases load on the server unnecessarily.

Reply:
We can create a separate url for create_UserProfile function and use the adapter (following link uses adapter for the same). 
https://stackoverflow.com/questions/27759407/django-allauth-redirect-after-social-signup
Or we can even shift the code for UserProfile creation to this adapter . Just an idea......not sure if it will work, as i have never used adapters.py in django